app/bundles/LeadBundle/Entity/CompanyRepository.php

```diff
@@ -0,0 +1,475 @@
+        $results = $q->execute()->fetchAll();
+
+        return $results;
+    }
+    /**
+     * Function to remove non custom field columns from an arrayed lead row
+     *
+     * @param array $r
+     */
+    protected function removeNonFieldColumns(&$r)
+    {
+        $baseCols = $this->getBaseColumns('Mautic\\LeadBundle\\Entity\\Company', true);
+        foreach ($baseCols as $c) {
+            unset($r[$c]);
+        }
+        unset($r['owner_id']);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getTableAlias()
+    {
+        return 'comp';
+    }
+
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function addCatchAllWhereClause(&$q, $filter)
+    {
+        return $this->addStandardCatchAllWhereClause($q, $filter, array(
+            'comp.companyname',
+            'comp.companydescription'
+        ));
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function addSearchCommandWhereClause(&$q, $filter)
+    {
+        return $this->addStandardSearchCommandWhereClause($q, $filter);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getSearchCommands()
+    {
+        return $this->getStandardSearchCommands();
+    }
+
+    /**
+     * Get a list of lists
+     *
+     * @param bool   $user
+     * @param string $alias
+     * @param string $id
+     *
+     * @return array
+     */
+    public function getCompanies($user = false, $id = '')
+    {
+        $q = $this->_em->getConnection()->createQueryBuilder();
+        static $companies = array();
+
+        if (is_object($user)) {
+            $user = $user->getId();
+        }
+
+        $key = (int) $user.$id;
+        if (isset($companies[$key])) {
+            return $companies[$key];
+        }
+
+        $q->select('comp.id, comp.companyname')
+            ->from(MAUTIC_TABLE_PREFIX .'companies', 'comp');
+
+        if (!empty($user)) {
+            $q->orWhere('comp.created_by = :user');
+            $q->setParameter('user', $user);
+        }
+
+        if (!empty($id)) {
+            $q->andWhere(
+                $q->expr()->neq('comp.id', $id)
+            );
+        }
+
+        $q->orderBy('comp.companyname');
+
+        $results = $q->execute()->fetchAll();
+
+        $companies[$key] = $results;
+
+        return $results;
+    }
+
+    /**
+     * Get a list of lists
+     *
+     * @param string $name
+     *
+     * @return array
+     */
+    public function getCompanyByName($companyName)
+    {
+        $q = $this->_em->getConnection()->createQueryBuilder();
+
+        if (!$companyName) {
+            return false;
+        }
+
+        $q->select('partial comp.{id, name}')
+            ->from('LeadBundle:Company', 'comp');
+
+        $q->andWhere(
+            $q->expr()->like('comp.name', $companyName)
+        );
+
+        $results = $q->execute()->fetchAll();
+
+        return $results;
+    }
+    /**
+    +     * Get a count of leads that belong to the company
```

Comment:
Please review these lines in your local editor. Something weird is going on. On the raw view of this page, it looks like you have extra `+` at the beginning of the next 6 lines.